src/framework/uicomponents/qml/MuseScore/UiComponents/StyledPopupView.qml

```diff
@@ -0,0 +1,128 @@
+import QtQuick 2.15
+import QtQuick.Controls 2.15
+import QtGraphicalEffects 1.0
+import MuseScore.UiComponents 1.0
+import MuseScore.Ui 1.0
+
+PopupView {
+    id: root
+
+    property bool opensUpward: false
+    property var arrowX: width / 2
+    property alias arrowHeight: arrow.height
+
+    property color borderColor: ui.theme.strokeColor
+    property color fillColor: ui.theme.backgroundPrimaryColor
+    readonly property int borderWidth: 1
+
+    closePolicy: PopupView.CloseOnPressOutsideParent
+
+    onAboutToShow: {
+        globalPos = mapToGlobal(x, y)
```

Comment:
can you move it to c++ class of popupView?

Reply:
added in-line comment with an explanation